cfme/common/vm_views.py

```diff
@@ -64,6 +64,65 @@ def InstanceEntity():  # noqa
     })
 
 
+class NonEditableTable(Table):
+    """Wigdet for non-editable table. used for selecting value"""
+    def fill(self, values):
+        """Clicks on item - fill by selecting required value"""
+        value = values.get('name', '<None>')
+        changed = False
+        if value != self.read():
+            changed = True
+        self.row(name=value).click()
+        return changed
+
+    @property
+    def currently_selected(self):
+        """Return Name of the selected row"""
+        selected = self.browser.elements(
+            ".//tr[@class='selected']/td[1]",
+            parent=self)
+        result = map(self.browser.text, selected)
+        if len(result) == 0:
+            self.logger.info('Nothing is currently selected')
+            return None
+        else:
+            return result[0]
+
+    def read(self):
+        return self.currently_selected
+
+    def read_content(self):
+        """This is a default Table.read() method for those who will need table content"""
```

Comment:
Is this copied from the original table class? You can do just `return super(NonEditableTable, self).read()` in this method to bridge this behaviour into this method.

Reply:
my bad. fixed